main/src/ca/uwaterloo/flix/language/ast/TokenKind.scala

```diff
@@ -238,6 +238,15 @@ sealed trait TokenKind {
     case _ => false
   })
 
+  /**
+    * Checks if this token is one of the [[TokenKind]]s that can validly appear after a doc-comment.
+    */
+  def isDocumentable: Boolean = this.isFirstDecl || (this match {
```

Comment:
Why is this "obviously" correct?

Reply:
I'll update the comment to 
```scala
/**
  * Checks if this token is one of the [[TokenKind]]s that can validly appear after a doc-comment.
  * Doc-comments may only annotate declarations an enum cases. 
  */
```
It's not obviously correct to do this, but that's the design we chose to have better guarantees about doc-comments.